docs/ref/forms/api.txt

```diff
@@ -260,9 +260,8 @@ precedence::
 
 .. method:: Form.get_initial_for_field(field, field_name)
 
-Use :meth:`~Form.get_initial_for_field()` to retrieve initial data for a form
-field. It retrieves data from :attr:`Form.initial` and :attr:`Field.initial`,
-in that order, and evaluates any callable initial values.
+While :meth:`~Form.get_initial_for_field()` exists, it is better to use
+:attr:`BoundField.initial` to retrieve the initial data for a form field.
```

Comment:
I'm not sure about this phrasing. I think it's better if the method is left in for people looking for it / expecting to find it, but at the same time, I think we want to steer people away from it.

Reply:
Hmmm. I'd likely just add the note below

```
:meth:`~Form.get_initial_for_field()` allows you to retrieve initial data for a
form field. It retrieves data from :attr:`Form.initial` and
:attr:`Field.initial`, in that order, and evaluates any callable initial
values.
  
Note that :attr:`BoundField.initial` is preferred, as it provides simpler access to the 
initial data for a form field.
```

🤔

Reply:
Okay, thanks, @carltongibson. I was more concerned about the potential buggy behavior of going through `get_initial_for_field()`, but I guess that can be fixed. I'm filing an issue about it now (alluded to in my ticket)...

Reply:
Happy to take suggestions on the `as it provides simpler access to the initial data for a form field.` — less might be more: `Note that :attr:`BoundField.initial` is preferred` and leave it at that? 🤔

I guess we could de-doc it, but there'll be a whole 15 years worth of people looking for it. …